cpp/include/cuvs/neighbors/cagra.cuh

```diff
@@ -29,397 +25,68 @@
- * @param[in] queries a device matrix view to a row-major matrix [n_queries, index->dim()]
- * @param[out] neighbors a device matrix view to the indices of the neighbors in the source dataset
- * [n_queries, k]
- * @param[out] distances a device matrix view to the distances to the selected neighbors [n_queries,
- * k]
- * @param[in] sample_filter a device filter function that greenlights samples for a given query
- */
-template <typename T, typename IdxT, typename CagraSampleFilterT>
-void search_with_filtering(raft::resources const& res,
-                           const search_params& params,
-                           const index<T, IdxT>& idx,
-                           raft::device_matrix_view<const T, int64_t, raft::row_major> queries,
-                           raft::device_matrix_view<IdxT, int64_t, raft::row_major> neighbors,
-                           raft::device_matrix_view<float, int64_t, raft::row_major> distances,
-                           CagraSampleFilterT sample_filter = CagraSampleFilterT())
-{
-  RAFT_EXPECTS(
-    queries.extent(0) == neighbors.extent(0) && queries.extent(0) == distances.extent(0),
-    "Number of rows in output neighbors and distances matrices must equal the number of queries.");
-
-  RAFT_EXPECTS(neighbors.extent(1) == distances.extent(1),
-               "Number of columns in output neighbors and distances matrices must equal k");
-  RAFT_EXPECTS(queries.extent(1) == idx.dim(),
-               "Number of query dimensions should equal number of dimensions in the index.");
-
-  using internal_IdxT   = typename std::make_unsigned<IdxT>::type;
-  auto queries_internal = raft::make_device_matrix_view<const T, int64_t, raft::row_major>(
-    queries.data_handle(), queries.extent(0), queries.extent(1));
-  auto neighbors_internal = raft::make_device_matrix_view<internal_IdxT, int64_t, raft::row_major>(
-    reinterpret_cast<internal_IdxT*>(neighbors.data_handle()),
-    neighbors.extent(0),
-    neighbors.extent(1));
-  auto distances_internal = raft::make_device_matrix_view<float, int64_t, raft::row_major>(
-    distances.data_handle(), distances.extent(0), distances.extent(1));
-
-  cagra::detail::search_main<T, internal_IdxT, CagraSampleFilterT, IdxT>(
-    res, params, idx, queries_internal, neighbors_internal, distances_internal, sample_filter);
-}
-
-/** @} */  // end group cagra
-
-}  // namespace cuvs::neighbors::cagra
-
-// TODO: Remove deprecated experimental namespace in 23.12 release
-namespace cuvs::neighbors::experimental::cagra {
-using cuvs::neighbors::cagra::build;
-using cuvs::neighbors::cagra::build_knn_graph;
-using cuvs::neighbors::cagra::optimize;
-using cuvs::neighbors::cagra::search;
-using cuvs::neighbors::cagra::sort_knn_graph;
-}  // namespace cuvs::neighbors::experimental::cagra
+// Using device and host_matrix_view avoids needing to typedef multiple mdspans based on accessors
+#define CUVS_INST_CAGRA_FUNCS(T, IdxT)                                             \
```

Comment:
This file should be an `hpp` since it's no longer compiling any kernels. The reason we used `cuh` for raft is because it's header-only and any headers that compile kernels should be marked as such (by convention).